enterprise/backend/test/metabase_enterprise/serialization/v2/storage_test.clj

```diff
@@ -150,3 +153,39 @@
                                                 "fields"    "Company__SLASH__organization website.yaml"))
                        (update :visibility_type keyword)
                        (update :base_type       keyword))))))))))
+
+(deftest yaml-sorted-test
+  (mt/with-empty-h2-app-db
+    (ts/with-temp-dpc [Card          c1 {:name "some card" :collection_id nil}
+                       Card          c2 {:name "other card" :collection_id nil}
+                       Dashboard     d1 {:name "some dash" :collection_id nil}
+                       DashboardCard _  {:card_id      (:id c1)
+                                         :dashboard_id (:id d1)}
+                       DashboardCard _  {:card_id      (:id c2)
+                                         :dashboard_id (:id d1)}]
+      (let [export (extract/extract nil)]
+        (with-redefs [spit (fn [fname yaml-data]
+                             (testing fname
+                               (doseq [coll (->> (yaml/parse-string yaml-data)
+                                                 (tree-seq coll? identity)
+                                                 (filter map?))]
+                                 (let [t     (-> (:serdes/meta coll) last :model)
+                                       order (when t
+                                               (get @@#'dump/serialization-order t))]
+                                   (if-not order
+                                     (is (= (not-empty (sort (keys coll)))
```

Comment:
Let's make sure that the set of things without a matching order is what we expect.

Or, better, let's intend to have a defined order for everything and assert `(is (some? order))` in all cases. One of the requirements for serdes v2 is to enforce with tests that any new models are properly handled, and a test like this which forces new models to get added to `serialization-order.edn` is a win.

Reply:
I've added a few models manually, but now I'm thinking if there is a good way to iterate through all defined models?..

Reply:
There are already lists for this in `metabase-enterprise.serialiazation.v2.models`, both the `exported-models` that exist at the top level and `inlined-models` that are nested in something else. The list of all models that are defined in the system is `metabase-enterprise.serialization.v2.entity-ids/toucan-models`.

I started writing a paragraph about adding more tests that serdes things are implemented for all models, but the right place for that is as a bug: #35540. Definitely not part of this PR!